src/marker.rs

```diff
@@ -357,6 +357,88 @@ fn parse_line_list_msg(
     lines
 }
 
+fn parse_sphere_msg(
+    msg: &rosrust_msg::visualization_msgs::Marker,
+    color: &tui::style::Color,
+    iso: &Isometry3<f64>,
+) -> Vec<Line> {
+    let mut lines: Vec<Line> = Vec::new();
+
+    let scale = &Point3::new(msg.scale.x,msg.scale.y,msg.scale.z);
+
+    let p1 = iso.transform_point(&Point3::new(-scale.x*0.5,0.0,0.0));
+    let p2 = iso.transform_point(&Point3::new(scale.x*0.5,0.0,0.0));
+    let p3 = iso.transform_point(&Point3::new(0.0,-scale.y*0.5,0.0));
+    let p4 = iso.transform_point(&Point3::new(0.0,scale.y*0.5,0.0));
+    let p5 = iso.transform_point(&Point3::new(0.0,0.0,-scale.z*0.5));
+    let p6 = iso.transform_point(&Point3::new(0.0,0.0,scale.z*0.5));
+
+
+    lines.push(Line {
+        x1: p1.x,
+        y1: p1.y,
+        x2: p2.x,
+        y2: p2.y,
+        color: *color,
+    });
+    lines.push(Line {
+        x1: p3.x,
+        y1: p3.y,
+        x2: p4.x,
+        y2: p4.y,
+        color: *color,
+    });
+    lines.push(Line {
+        x1: p5.x,
+        y1: p5.y,
+        x2: p6.x,
+        y2: p6.y,
+        color: *color,
+    });
+
+    let n = 20;
+    let step = (2.0*PI)/(n as f64);
+    for i  in 0..n {
+        let ifl = i as f64;
+        let pa = iso.transform_point(&Point3::new(0.5*scale.x*((ifl*(step)).sin()),0.5*scale.y*((ifl*(step)).cos()),0.0));
```

Comment:
These lines are too long to parse them effectively in brains (at least mine).  Some line breaks and documentation will make it easier

Reply:
Done